### PR TITLE
[core][runtime_env] Allow full path in conda runtime env.

### LINKED ISSUE #44373: specifying conda runtime_env using fullpath no longer works
### What happened + What you expected to happen

In earlier version of ray (2.6.3), I was able to specify full path to conda environment to select a specific conda environment. In latest version (2.9.3), I get an error 

`
  File "/usr/conda/lib/python3.10/site-packages/ray/_private/runtime_env/conda.py", line 348, in _create
    raise ValueError(
ValueError: The given conda environment '/usr/conda' from the runtime env {'_ray_commit': '62655e11ed76509b78654b60be67bc59f8f3460a', 'conda': '/usr/conda'} doesn't exist from the output of `conda env list --json`. You can only specify an env that already exists. Please make sure to create an env /usr/conda 
`

"/usr/conda" is present in `conda env list --json`.  This is broken by this PR

https://github.com/ray-project/ray/commit/69ed38b0efbb30576731c84d2886610b1d63ae7d

`envs = [Path(env).name for env in conda_env_list]` line is only checking the last path component and not the full path.

I tried specifying just the last path component but that also does not work. Conda environments created using -p prefix are no longer working.

### Versions / Dependencies

Ray: 2.9.3
Python: 3.10.13

### Reproduction script

```
import ray
ray.init(runtime_env={"conda": "/usr/conda"})
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by rynewang**
#34956 fixed a conda hanging issue by checking existence of the conda env name. However it only checks by conda env name, not path - if the user specifies an absolute path, it no longer works. This is an regression because it should support full paths without a problem.

This PR reads conda CLI for the full paths and names. A special treatment is done for `base` which is just the conda base prefix; and others are base conda_prefix/envs/name.

Adds tests for both `base` and full path of it; and also name and full paths for non base envs.

Fixes #44373 .

### CODE REVIEW

**Comment by jjyao** [comment]
```
conda (dict | str): Either (1) a dict representing the conda environment YAML, (2) a string containing the path to a local [conda “environment.yml”](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually) file, or (3) the name of a local conda environment already installed on each node in your cluster (e.g., "pytorch_p36")
```

from https://docs.ray.io/en/latest/ray-core/handling-dependencies.html doesn't mention that we support full path only the name.


If we want to support it, we should also update our doc.

**Comment by rynewang** [comment]
Doc updated

**Comment by rynewang** [comment]
@angelinalg : added a clause to the doc about how to specify `conda` runtime env. Previously you can set a env name like `pytorch_p36`, now you can also set an absolute path of it like `"/home/youruser/anaconda3/envs/pytorch_p36"`. Also updated some single backticks to double backticks to be consistent with rest of doc.

File: python/ray/_private/runtime_env/conda_utils.py [hunk 1]
```diff
@@ -173,6 +173,33 @@ def get_conda_env_list() -> list:
     return envs
 
 
+def get_conda_info_json() -> dict:
```
Comment (hongchaodeng) [inline python/ray/_private/runtime_env/conda_utils.py:176]: Can you add a link or an example of the return value?

**Review by hongchaodeng — APPROVED** [review]
LGTM after nit

File: python/ray/_private/runtime_env/conda_utils.py (hunk 1 above)
Comment (rynewang) [reply python/ray/_private/runtime_env/conda_utils.py:176]: done

**Review by can-anyscale — APPROVED** [review]

**Review by jjyao — APPROVED** [review]

**Pull request merged**